### ISSUE TITLE
No LEDS working - CM MasterKeys Pro L 

### ISSUE THREAD

**Issue by platcham** [SpoinkyNL/Artemis#203]
Hello!

I cannot seem to get Artemis to work at all. Whenever I enable Artemis, my keyboard just goes dark and does not show any effect. I have tried this on both my desktop and laptop, with similar results. Below is my log file, which seems to be giving an error with the Lua wrapper, but I don't know if that's related or not. 

Thanks for the help! 

Here is my most recent error log:

> 2016-11-29 11:38:21.9176|FATAL|Artemis.Utilities.Logging|INFO: Set log level to Info 
2016-11-29 11:38:22.1634|INFO|Artemis.Managers.DeviceManager|Intialized DeviceManager with 9 device providers 
2016-11-29 11:38:22.8258|INFO|Artemis.Managers.EffectManager|Intialized EffectManager 
2016-11-29 11:38:22.8258|INFO|Artemis.Managers.LoopManager|Intialized LoopManager 
2016-11-29 11:38:22.8258|INFO|Artemis.Managers.ProfileManager|Intialized ProfileManager 
2016-11-29 11:38:22.8368|INFO|Artemis.Utilities.DataReaders.PipeServer|Opened named pipe 'artemis' 
2016-11-29 11:38:22.8368|INFO|Artemis.Managers.MainManager|Intialized MainManager 
2016-11-29 11:38:22.8368|INFO|Artemis.Utilities.Updater|Update check enabled: True 
2016-11-29 11:38:22.8368|INFO|Artemis.Managers.MainManager|Artemis version 1.5.0.3 is ready! 
2016-11-29 11:38:23.7271|INFO|Artemis.Utilities.Updater|Update check complete 
2016-11-29 11:38:26.4089|ERROR|Artemis.Profiles.Lua.LuaWrapper|[Default-LUA]: Error: chunk_2:(1,0-4): unexpected symbol near 'true' MoonSharp.Interpreter.SyntaxErrorException: unexpected symbol near 'true'
   at MoonSharp.Interpreter.Tree.Expression.PrefixExp(ScriptLoadingContext lcontext)
   at MoonSharp.Interpreter.Tree.Expression.PrimaryExp(ScriptLoadingContext lcontext)
   at MoonSharp.Interpreter.Tree.Statement.CreateStatement(ScriptLoadingContext lcontext, Boolean& forceLast)
   at MoonSharp.Interpreter.Tree.Statements.CompositeStatement..ctor(ScriptLoadingContext lcontext)
   at MoonSharp.Interpreter.Tree.Statements.ChunkStatement..ctor(ScriptLoadingContext lcontext)
   at MoonSharp.Interpreter.Tree.Fast_Interface.Loader_Fast.LoadChunk(Script script, SourceCode source, ByteCode bytecode)
   at MoonSharp.Interpreter.Script.LoadString(String code, Table globalTable, String codeFriendlyName)
   at Artemis.Profiles.Lua.LuaWrapper.SetupLua(ProfileModel profileModel, KeyboardProvider keyboardProvider)
2016-11-29 11:38:26.7820|ERROR|Artemis.Profiles.Lua.LuaWrapper|[Visualizer duplicate-LUA]: Error: chunk_10:(26,8-87): cannot access field Color of userdata<Artemis.Modules.Effects.WindowsProfile.WindowsProfileDataModel> MoonSharp.Interpreter.ScriptRuntimeException: cannot access field Color of userdata<Artemis.Modules.Effects.WindowsProfile.WindowsProfileDataModel>
   at MoonSharp.Interpreter.Execution.VM.Processor.ExecIndex(Instruction i, Int32 instructionPtr)
   at MoonSharp.Interpreter.Execution.VM.Processor.Processing_Loop(Int32 instructionPtr)
   at MoonSharp.Interpreter.Execution.VM.Processor.Call(DynValue function, DynValue[] args)
   at MoonSharp.Interpreter.Interop.EventMemberDescriptor.DispatchEvent(Object sender, Object o01, Object o02, Object o03, Object o04, Object o05, Object o06, Object o07, Object o08, Object o09, Object o10, Object o11, Object o12, Object o13, Object o14, Object o15, Object o16)
   at MoonSharp.Interpreter.Interop.EventMemberDescriptor.<>c__DisplayClass22_0.<CreateDelegate>b__2(Object o1, Object o2)
   at Artemis.Profiles.Lua.Events.LuaEventsWrapper.LuaInvoke(ProfileModel profileModel, Action action)

**Comment by SpoinkyNL** [comment]
Hey, looks weird, the LUA wrapper might indeed be breaking it.

Could you try creating a new profile and just adding a layer that covers your entire keyboard and see if that lights it up?

**Comment by platcham** [comment]
No dice. Still just a dark keyboard. And now I'm not getting any errors in the error log. 

Here's how I have the new layer set up. Anything weird here that could be tripping it up?
![capture](https://cloud.githubusercontent.com/assets/22689390/20727740/5e1a209c-b640-11e6-93f0-4b343a09f6a8.PNG)

**Comment by SpoinkyNL** [comment]
Hmm that's strange, do you have a different PC on hand on which you could try it?

**Comment by platcham** [comment]
Yep, I tried it with my laptop as well. Same results. It'll light up with its built in backlighting when plugged into the computer, but as soon as Artemis takes control of the keyboard, it goes completely dark.

**Comment by SpoinkyNL** [comment]
Hmm that's weird, I tried the latest release and it works well for me. It might be worth installing [Visual Studio C++ Redistibutable 2015](https://www.microsoft.com/en-us/download/details.aspx?id=48145) be sure to grab the x64 version  
And  [Visual C++ Redistributable for Visual Studio 2012 Update 4 - 64 bit](https://www.microsoft.com/en-au/download/confirmation.aspx?id=30679) as well

Let me know if that helps :)

**Comment by platcham** [comment]
I'll try installing those when I get back from work.

Thanks for all the help by the way!

**Comment by platcham** [comment]
Unfortunately, those did not help. I actually already had both installed, so I tried uninstalling and reinstalling them. Still a blank keyboard when any effects are enabled.

**Comment by SpoinkyNL** [comment]
No problem, I hope we can figure it out :)

Could you try any of the custom effects here to see if they work? 
http://makerhub.coolermaster.com/custom-lighting/

**Comment by platcham** [comment]
I have solved the issue! Half of their examples worked, half didn't, which I thought was weird. Eventually I found out that I just needed to update the firmware on my keyboard to the latest version. Artemis now works as expected! 

On a side note, I found that I experienced severe flickering with Artemis if any of the built in LED profiles were enabled. Setting a blank profile solved this issue. 

Thanks for the support and the awesome program!
